sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexAsyncClient.java

```diff
@@ -0,0 +1,732 @@
+     * @param suggestOptions suggest options
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @return A {@link PagedFluxBase} that iterates over {@link SuggestResult} objects
+     * and provides access to the {@link SuggestPagedResponse} object for each page containing
+     * HTTP response and coverage information.
+     */
+    public PagedFluxBase<SuggestResult, SuggestPagedResponse> suggest(String searchText,
+                                                                      String suggesterName,
+                                                                      SuggestOptions suggestOptions,
+                                                                      RequestOptions requestOptions) {
+        SuggestRequest suggestRequest = this.createSuggestRequest(searchText,
+            suggesterName, SuggestOptionsHandler.ensureSuggestOptions(suggestOptions));
+        return new PagedFluxBase<>(
+            () -> withContext(context -> this.suggestFirst(requestOptions, suggestRequest, context)));
+    }
+
+    PagedFluxBase<SuggestResult, SuggestPagedResponse> suggest(String searchText,
+                                                               String suggesterName,
+                                                               SuggestOptions suggestOptions,
+                                                               RequestOptions requestOptions,
+                                                               Context context) {
+        SuggestRequest suggestRequest = this.createSuggestRequest(searchText,
+            suggesterName, SuggestOptionsHandler.ensureSuggestOptions(suggestOptions));
+        return new PagedFluxBase<>(
+            () -> this.suggestFirst(requestOptions, suggestRequest, context));
+    }
+
+    /**
+     * Sends a batch of document actions to the Azure Cognitive Search index.
+     *
+     * @param batch batch of documents to send to the index with the requested action
+     * @return document index result
+     */
+    public Mono<IndexDocumentsResult> index(IndexBatch<?> batch) {
+        return this.indexWithResponse(batch)
+            .map(Response::getValue);
+    }
+
+    /**
+     * Sends a batch of document actions to the Azure Cognitive Search index.
+     *
+     * @param batch batch of documents to send to the index with the requested action
+     * @return a response containing the document index result
+     */
+    public Mono<Response<IndexDocumentsResult>> indexWithResponse(IndexBatch<?> batch) {
+        return withContext(context -> indexWithResponse(batch, context));
+    }
+
+    Mono<Response<IndexDocumentsResult>> indexWithResponse(IndexBatch<?> batch, Context context) {
+        return restClient.documents()
+            .indexWithRestResponseAsync(batch, context)
+            .handle((res, sink) -> {
+                if (res.getStatusCode() == 207) {
```

Comment:
Use a constant `MULTI_STATUS_RESPONSE` instead of 207.

Reply:
@navalev Looks like this was missed.